src/qpython_imageprovider.cpp

```diff
@@ -163,11 +167,61 @@ QPythonImageProvider::requestImage(const QString &id, QSize *size, const QSize &
         goto cleanup;
     }
 
+    if (format < 0) {
+        switch (format) {
+        case PYOTHERSIDE_IMAGE_FORMAT_ENCODED: {
+            // Pixel data is actually encoded image data that we need to decode
+            img.loadFromData((const unsigned char*)PyByteArray_AsString(pixels), PyByteArray_Size(pixels));
+            break;
+        }
+        case PYOTHERSIDE_IMAGE_FORMAT_SVG: {
+            // Convert the Python byte array to a QByteArray
+            QByteArray svgDataArray(PyByteArray_AsString(pixels), PyByteArray_Size(pixels));
+
+            // Load the SVG data to the SVG renderer
+            QSvgRenderer renderer(svgDataArray);
+
+            // Handle width, height or both not being set
+            if (width < 0 && height < 0) {
+                // Both Width and Height have not been set - use the defaults from the SVG data
+                // (each SVG image has a default size)
+                // NOTE: we get a -1,-1 requestedSize only if sourceSize is not set at all,
+                //       if either width or height is set then the other one is 0, not -1
+                width = renderer.defaultSize().width();
+                height = renderer.defaultSize().height();
+            } else { // At least width or height is valid
+                if (width <= 0) {
+                    // Width is not set, use default width scaled according to height to keep
+                    // aspect ratio
+                    if (renderer.defaultSize().height() != 0) {  // Protect from division by zero
+                        width = (float)renderer.defaultSize().width()*((float)height/(float)renderer.defaultSize().height());
+                    }
+                }
+
+                if (height <= 0) {
+                    // Height is not set, use default height scaled according to width to keep
+                    // aspect ratio
+                    if (renderer.defaultSize().width() != 0) {  // Protect from division by zero
+                        height = (float)renderer.defaultSize().height()*((float)width/(float)renderer.defaultSize().width());
+                    }
+                }
+            }
+
+            // Pixel data is actually SVG image data that we need to render at correct size
+            img = QImage(width, height, QImage::Format_ARGB32);
+
+            // According to the documentation an empty QImage needs to be "flushed" before
+            // being used with QPainter to prevent rendering artifacts from showing up
+            img.fill(Qt::transparent);
 
-    if (format == -1) {
-        // Pixel data is actually encoded image data that we need to decode
-        img.loadFromData((const unsigned char*)PyByteArray_AsString(pixels),
-                PyByteArray_Size(pixels));
+            // Paints the rendered SVG to the QImage instance
+            QPainter painter(&img);
+            renderer.render(&painter);
+            break;
+        }
+        default:
```

Comment:
Maybe issue a `qWarning()` that an unknown format has been passed and will not be handled.